src/api/services/rng.py

```diff
@@ -4,16 +4,16 @@
 
 
 class RNG:
-    def __init__(self, weights: Union[Rarity, list[int]] = None):
-        if weights is None:
-            self.weights = Rarity.Poor.value
+    def __init__(self, rarity_weights: Union[Rarity, list[int]] = None):
+        if rarity_weights is None:
+            self.weights = Rarity.Poor.weights
 
         else:
-            if isinstance(weights, Rarity):
-                self.weights = weights.value
+            if isinstance(rarity_weights, Rarity):
+                self.weights = rarity_weights.weights
 
-            elif isinstance(weights, list):
-                self.weights = weights
+            elif isinstance(rarity_weights, list):
+                self.weights = rarity_weights
 
         self.rewards: list[Rarity] = [
```

Comment:
Pretty sure you can just `list(Rarity)` to make the enum a list

Reply:
we need to do different things if its an enum compared to if its a list
```py
            if isinstance(rarity_weights, Rarity):
                self.weights = rarity_weights.weights

            elif isinstance(rarity_weights, list):
                self.weights = rarity_weights
```